stream-chat-android-client/src/main/java/io/getstream/chat/android/client/network/NetworkStateProvider.kt

```diff
@@ -56,8 +56,9 @@ internal class NetworkStateProvider(private val connectivityManager: Connectivit
 
     fun subscribe(listener: NetworkStateListener) {
         listeners = listeners + listener
-        connectivityManager.registerNetworkCallback(NetworkRequest.Builder().build(), callback)
-        isRegistered.set(true)
+        if (isRegistered.compareAndSet(false, true)) {
```

Comment:
Only register our callback if we didn't do it previously (And it is registered yet)